xclim/sdba/measures.py

```diff
@@ -490,15 +490,12 @@ def _taylordiagram(
         [ref_std, sim_std, corr],
         coords=coords,
         dims=dims,
-        attrs={"correlation_type": "Pearson correlation coefficient"},
+        attrs={
+            "correlation_type": "Pearson correlation coefficient",
+            "units": ref_std.units,
```

Comment:
Do we know that `ref` will always have `units`?

Reply:
If it gets to this point, yes. I tried passing a DataArray without units and `units2pint` in `preprocess_and_checks` raises a `KeyError`  (upstream of this function).